exla/lib/exla/defn.ex

```diff
@@ -702,6 +702,25 @@ defmodule EXLA.Defn do
     EXLA.Op.sort(tensor, comp, dimension)
   end
 
+  defp to_operator(:argsort, [tensor, opts, comparator], %{type: type, shape: shape}, state) do
+    dimension = opts[:axis]
+
+    # Grow the comparator to arity 4 because argsort uses
+    # variadic_sort underneath
+    path = [Access.key!(:data), Access.key!(:args)]
+    [[arg_1, arg_2], expr, fun] = get_in(comparator, path)
+
+    comparator_4 =
+      put_in(comparator, path, [
+        [arg_1, arg_2, arg_1, arg_2],
+        expr,
+        fn x, y, _, _ -> fun.(x, y) end
+      ])
```

Comment:
I'm not sure if there's a better way to achieve this. Basically, the comparison function needed for variadic_sort always has 2*num_args arity, so I needed to extend the arity-2 comparator to an arity-4 one which only uses the first 2 args (which are the current comparison elements for the first of the args passed to sort)

Reply:
I think you can wrap that function inside the existing function. Something like this:

```elixir
    # Grow the comparator to arity 4 because argsort uses
    # variadic_sort underneath
    [[arg1, arg2], _expr, fun] = comparator.data.args

    comparator_4 = Expr.fun([arg1, arg2, arg1, arg2], fn x, y, _, _ -> fun.(x, y) end)
```